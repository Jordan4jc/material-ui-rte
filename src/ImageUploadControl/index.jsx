import React from "react"
import ButtonGroup from "@material-ui/core/ButtonGroup"
import Button from "@material-ui/core/Button"
import AddPhotoAlternateIcon from "@material-ui/icons/AddPhotoAlternate"

const ImageUploadControl = props => {
  return (
    <ButtonGroup variant="contained" size="small">
      <Button onClick={() => props.onClick()}>
        <AddPhotoAlternateIcon />
      </Button>
    </ButtonGroup>
  )
}

export default ImageUploadControl
