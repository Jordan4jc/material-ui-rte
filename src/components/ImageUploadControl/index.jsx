import React, { Fragment } from "react"
import ButtonGroup from "@material-ui/core/ButtonGroup"
import Button from "@material-ui/core/Button"
import AddPhotoAlternateIcon from "@material-ui/icons/AddPhotoAlternate"

const ImageUploadControl = props => {
  const handleChange = files => {
    let file = files[0]
    file.preview = URL.createObjectURL(files[0])
    props.onFile(file)
    setTimeout(() => {
      document.getElementById("file-input").value = null
    }, 50)
  }
  return (
    <Fragment>
      <ButtonGroup variant="contained" size="small">
        <Button onClick={() => document.getElementById("file-input").click()}>
          <AddPhotoAlternateIcon />
        </Button>
      </ButtonGroup>
      <input
        type="file"
        style={{
          display: "none"
        }}
        onChange={e => handleChange(e.target.files)}
        name="file"
        id="file-input"
        accept="image/*"
      />
    </Fragment>
  )
}

export default ImageUploadControl
