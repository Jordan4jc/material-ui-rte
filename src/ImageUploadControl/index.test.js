import React from "react"
import { shallow } from "enzyme"
import ImageUploadControl from "./index"

describe("ImageUploadControl", () => {
  it("Should render image upload control", () => {
    const component = shallow(<ImageUploadControl />)

    expect(component).toMatchSnapshot()
  })
})
