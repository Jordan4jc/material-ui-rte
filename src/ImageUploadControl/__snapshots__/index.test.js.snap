// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DividerControl Should render alignment control 1`] = `ShallowWrapper {}`;

exports[`ImageUploadControl Should render image upload control 1`] = `ShallowWrapper {}`;
